vector.go

```diff
@@ -252,163 +267,52 @@ func (vec *vector) getChildVectors(vector C.duckdb_vector) {
 	}
 }
 
-func (vec *vector) setNull(rowIdx C.idx_t) {
-	C.duckdb_vector_ensure_validity_writable(vec.duckdbVector)
-	mask := C.duckdb_vector_get_validity(vec.duckdbVector)
-	C.duckdb_validity_set_row_invalid(mask, rowIdx)
-
-	if vec.duckdbType == C.DUCKDB_TYPE_STRUCT {
-		for i := 0; i < len(vec.childVectors); i++ {
-			vec.childVectors[i].setNull(rowIdx)
-		}
-	}
-}
-
-func setPrimitive[T any](vec *vector, rowIdx C.idx_t, val any) {
-	if val == nil {
-		vec.setNull(rowIdx)
-		return
-	}
-
-	ptr := C.duckdb_vector_get_data(vec.duckdbVector)
-	xs := (*[1 << 31]T)(ptr)
-	xs[rowIdx] = val.(T)
-}
-
-func (vec *vector) setCString(rowIdx C.idx_t, val any) {
-	if val == nil {
-		vec.setNull(rowIdx)
-		return
-	}
-
-	var str string
-	switch vec.duckdbType {
-	case C.DUCKDB_TYPE_VARCHAR:
-		str = val.(string)
-	case C.DUCKDB_TYPE_BLOB:
-		str = string(val.([]byte)[:])
-	}
-
-	// This setter also writes BLOBs.
-	cStr := C.CString(str)
-	C.duckdb_vector_assign_string_element_len(vec.duckdbVector, rowIdx, cStr, C.idx_t(len(str)))
-	C.free(unsafe.Pointer(cStr))
-}
-
-func (vec *vector) setTime(rowIdx C.idx_t, ticks int64) {
-	var ts C.duckdb_timestamp
-	ts.micros = C.int64_t(ticks)
-	setPrimitive[C.duckdb_timestamp](vec, rowIdx, ts)
-}
-
-func (vec *vector) setDate(rowIdx C.idx_t, days int32) {
-	var date C.duckdb_date
-	date.days = C.int32_t(days)
-	setPrimitive[C.duckdb_date](vec, rowIdx, date)
-}
-
-func (vec *vector) setList(rowIdx C.idx_t, val any) {
-	if val == nil {
-		vec.setNull(rowIdx)
-		return
-	}
-
-	v := val.([]any)
-	childVectorSize := C.duckdb_list_vector_get_size(vec.duckdbVector)
-
-	// Set the offset and length of the list vector using the current size of the child vector.
-	listEntry := C.duckdb_list_entry{
-		offset: C.idx_t(childVectorSize),
-		length: C.idx_t(len(v)),
-	}
-	setPrimitive[C.duckdb_list_entry](vec, rowIdx, listEntry)
-
-	newLength := C.idx_t(len(v)) + childVectorSize
-	C.duckdb_list_vector_set_size(vec.duckdbVector, newLength)
-	C.duckdb_list_vector_reserve(vec.duckdbVector, newLength)
-
-	// Insert the values into the child vector.
-	childVector := vec.childVectors[0]
-	for i, e := range v {
-		offset := C.idx_t(i) + childVectorSize
-		childVector.fn(&childVector, offset, e)
-	}
-}
-
-func (vec *vector) setStruct(rowIdx C.idx_t, val any) {
-	if val == nil {
-		vec.setNull(rowIdx)
-		return
-	}
-	m := val.(map[string]any)
-
-	for i := 0; i < len(vec.childVectors); i++ {
-		childVector := vec.childVectors[i]
-		childName := vec.childNames[i]
-		childVector.fn(&childVector, rowIdx, m[childName])
-	}
-}
-
 func initPrimitive[T any](vec *vector, duckdbType C.duckdb_type) {
-	vec.fn = func(vec *vector, rowIdx C.idx_t, val any) {
+	vec.setFn = func(vec *vector, rowIdx C.idx_t, val any) {
+		vec.size++
```

Comment:
This seems wrong, if we start with an empty vector, and set row `50` to x, the size should be 50 (right?). If we do it like this the size would be 1, and we would have one row with an unset value at 1.

Reply:
Yes, that's a fair argument. I'll create a fix PR. Sorry for not tagging you on this PR! However, we always need to set the maximum. Now that I think about this, we probably want to decouple this from the vector entirely and change `func (chunk *DataChunk) setSize(size int)` to take an argument.

Reply:
I don't seem to have gotten a mail about this, must have gotten lost in the sea of spam github sends about failed CI. But thanks for the response! I'll rebase my code sometime after next week

Reply:
No worries - :)